packages/token-sdk/src/metadata/client/metaplex-client.ts

```diff
@@ -0,0 +1,215 @@
+import { PublicKey } from "@solana/web3.js";
+import fetch from "isomorphic-unfetch";
+import invariant from "tiny-invariant";
+
+const METADATA_PROGRAM_ID = new PublicKey("metaqbxxUerdq28cj1RbAWkYQm3ybzjb6a8bt518x1s");
+
+interface Creator {
+  address: PublicKey;
+  verified: boolean;
+  share: number;
+}
+
+interface Collection {
+  verified: boolean;
+  key: PublicKey;
+}
+
+interface Uses {
+  useMethod: number;
+  remaining: BigInt;
+  total: BigInt;
+}
+
+interface OnChainMetadataPrefix {
+  key: number;
+  updateAuthority: PublicKey;
+  mint: PublicKey;
+  name: string;
+  symbol: string;
+  uri: string;
+  sellerFeeBasisPoints: number;
+}
+
+interface OnChainMetadataCreators {
+  creators: Creator[];
+}
+
+interface OnChainMetadataSuffix {
+  primarySaleHappened: boolean;
+  isMutable: boolean;
+  editionNonce: number | null;
+  tokenStandard: number | null;
+  collection: Collection | null;
+  uses: Uses | null;
+}
+
+export type OnChainMetadata = OnChainMetadataPrefix & OnChainMetadataCreators & OnChainMetadataSuffix
+
+export interface OffChainMetadata {
+  name?: string;
+  symbol?: string;
+  description?: string;
+  image?: string;
+}
+
+export interface MetaplexClient {
+  getMetadataAddress(mint: PublicKey): PublicKey;
+  parseOnChainMetadata(mint: PublicKey, buffer: Buffer | Uint8Array): OnChainMetadata | null;
+  getOffChainMetadata(metadata: OnChainMetadata): Promise<OffChainMetadata | null>;
+}
+
+export class MetaplexHttpClient implements MetaplexClient {
+
+  getMetadataAddress(mint: PublicKey): PublicKey {
+    const seeds = [Buffer.from("metadata"), METADATA_PROGRAM_ID.toBuffer(), mint.toBuffer()];
+    return PublicKey.findProgramAddressSync(seeds, METADATA_PROGRAM_ID)[0];
+  }
+
+  parseOnChainMetadata(mint: PublicKey, data: Uint8Array | Buffer): OnChainMetadata | null {
```

Comment:
Nice! Could you add a comment or two that could be helpful for the details of parsing for future reference?

For example was able to locate the Metadata struct in https://github.com/metaplex-foundation/mpl-token-metadata/blob/main/programs/token-metadata/program/src/state/metadata.rs#L63. Anything in the `umi` repo would be helpful as well.